from snakemake.utils import validate
import pandas as pd
import os

configfile: "config/config.yml"

# ----- start execution of the rules ----- #

rule sample_pangenome:
  input:
    pangenome=config["pangenome_roary"],
    genes=config["pangenome_genes"]
  output:
    "out/sample_nuc.faa"
  threads: 4
  conda: "envs/eggnog-mapper.yml"
  log: "out/logs/sample_nuc.log"
  shell:
    """
    python workflow/scripts/sample_pangenome.py {input.pangenome} {input.genes} \
           --nucleotide > {output}
    """

rule blast_pangenome_vags:
  input:
    query="db/virulence.fasta",
    nucl="out/sample_nuc.faa"
  output:
    "out/sample.blast.txt"
  threads: 4
  log: "out/logs/blast.log"
  conda: "envs/blast.yml"
  shell:
    """
    blastn -query {input.query} -out {output} -subject {input.nucl} -outfmt "6 qseqid sseqid pident qlen slen length nident"
    """
